src/Compiler/TypedTree/CompilerGlobalState.fs

```diff
@@ -92,12 +64,10 @@ type internal CompilerGlobalState () =
 type Unique = int64
 
 //++GLOBAL MUTABLE STATE (concurrency-safe)
-let newUnique =
-    let i = ref 0L
-    fun () -> System.Threading.Interlocked.Increment i
+let mutable private uniqueCount = 0L
```

Comment:
Changed to remove indirection (getting rid of closure and .Invoke() call) in the generated code.